zubhub_backend/zubhub/creators/views.py

```diff
@@ -91,10 +91,16 @@ class UserProfileAPIView(RetrieveAPIView):
     throttle_classes = [GetUserRateThrottle, SustainedRateThrottle]
 
     def get_serializer_class(self):
-        if self.kwargs.get("username") == self.request.user.username:
-            return CreatorSerializer
-        else:
+        if self.request and self.kwargs.get("username") != self.request.user.username:
             return CreatorMinimalSerializer
+        else:
+            return CreatorSerializer
+
+    """
```

Comment:
The docstring below should be a part of the docstring at the beginning of the class, not here